runtime/jcl/common/jdk_jfr_internal_JVM_common.cpp

```diff
@@ -350,8 +350,21 @@ Java_jdk_jfr_internal_JVM_getTypeId(JNIEnv *env, jobject obj, jclass clazz)
 Java_jdk_jfr_internal_JVM_getTypeId__Ljava_lang_Class_2(JNIEnv *env, jobject obj, jclass clazz)
 #endif /* JAVA_SPEC_VERSION == 11 */
 {
-	// TODO: implementation
-	return 0;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	j9object_t classObject = J9_JNI_UNWRAP_REFERENCE(clazz);
+	J9Class *ramClass = J9VMJAVALANGCLASS_VMREF(currentThread, classObject);
+	J9UTF8 *className = J9ROMCLASS_CLASSNAME(ramClass->romClass);
+
+	jlong result = vmFuncs->getTypeIDUTF8(currentThread, className, FALSE);
```

Comment:
Is this good enough for uniqueness? Class names are only unique within their class loader.